File: fix_lint.py

import os
import re

def fix_empty_interfaces(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(".tsx") or file.endswith(".ts"):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Regex to find: export interface X extends Y {}
                # Replace with: export type X = Y
                # Pattern needs to handle multiline potentially, but usually these are single line in shadcn
                
                # Pattern: export interface Name extends Extension {}
                pattern = r'export interface (\w+) extends ([^{]+) \{\}'
                
                if re.search(pattern, content):
                    print(f"Fixing {file}...")
                    new_content = re.sub(pattern, r'export type \1 = \2', content)
                    
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)

if __name__ == "__main__":
    target_dir = r"c:\Users\braza\Downloads\Leads control\src\components\ui"
    fix_empty_interfaces(target_dir)

File: fix_lint.js

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const targetDir = path.join(__dirname, 'src', 'components', 'ui');

function fixEmptyInterfaces(directory) {
    if (!fs.existsSync(directory)) {
        console.log(`Directory not found: ${directory}`);
        return;
    }

    const files = fs.readdirSync(directory);

    files.forEach(file => {
        const filepath = path.join(directory, file);
        const stat = fs.statSync(filepath);

        if (stat.isDirectory()) {
            fixEmptyInterfaces(filepath);
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            let content = fs.readFileSync(filepath, 'utf-8');

            // Regex to find: export interface X extends Y {}
            // Replace with: export type X = Y
            const pattern = /export interface (\w+) extends ([^{]+) \{\}/g;

            if (pattern.test(content)) {
                console.log(`Fixing ${file}...`);
                const newContent = content.replace(pattern, 'export type $1 = $2');
                fs.writeFileSync(filepath, newContent, 'utf-8');
            }
        }
    });
}

console.log('Starting lint fix...');
fixEmptyInterfaces(targetDir);
console.log('Done.');
